### PR TITLE
Clean up cleartask; update to latest django-q fork.

### PR DESCRIPTION
**Pull request by aronasorman**
Just rewrote @indirectlylit's implementation to be more Djangonic. And updated django-q to the latest version, to store both the raw result object, and to save pending tasks in memory.

### CODE REVIEW

File: kolibri/core/assets/src/api-resources/task.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ class TaskResource extends Resource {
 
 // TODO: switch to Model.delete()
   clearTask(taskId) {
-    const clientObj = { path: this.clearTaskUrl(), entity: { id: taskId } };
+    const clientObj = { path: this.clearTaskUrl(), entity: { task_id: taskId } };
```
Comment (jamalex) [inline kolibri/core/assets/src/api-resources/task.js:30]: I thought we were thinking to just be more RESTful and do this as a `DELETE` on `/task/555/`? Ah, I guess that's the TODO in the comment above this? (Note: @benjaoming has banned TODO's in the code :D -- opening issues will ensure it gets tracked and triaged -- or, in this case, should be a fairly simple change)

File: kolibri/tasks/api.py [hunk 2]
```diff
@@ -101,16 +101,19 @@ def startlocalexport(self, request):
...
+
+        if 'task_id' not in request.data:
+            raise serializers.ValidationError("The 'task_id' field is required.")
+
+        task_id = request.data['task_id']
+
+        # we need to decrypt tasks first to get their real task_id. Hence why this python-side task_id retrieval and deletion.
+        [taskitem.delete() for taskitem in OrmQ.objects.all() if taskitem.task()["id"] == task_id]
+
+        Task.objects.filter(pk=task_id).delete()
```
Comment (jamalex) [inline kolibri/tasks/api.py:115]: Given line 113, and the comment above it, what does this do?

Comment (aronasorman) [reply kolibri/tasks/api.py:115]: This removes the results -- `Task` is for the task results (where the progress data is stored), `OrmQ` is for the queues.

**Comment by indirectlylit** [comment]
is this getting additional changes before merging in for the bug bash?

**Comment by aronasorman** [comment]
This is done code-wise. Unfortunately I need to fix the tests before this can get merged to master.

**Pull request merged**